src/dune_rules/lib.mli

```diff
@@ -113,23 +126,27 @@ module DB : sig
       [all] returns the list of names of libraries available in this database. *)
   val create
     :  parent:t option
-    -> resolve:(Lib_name.t -> Resolve_result.t Memo.t)
-    -> all:(unit -> Lib_name.t list Memo.t)
+    -> resolve_name:(Lib_name.t -> Resolve_result.With_multiple_results.t Memo.t)
+    -> resolve_sentinel:(Lib_info.Sentinel.t -> Resolve_result.t Memo.t)
+    -> all:(unit -> Lib_info.Sentinel.t list Memo.t)
     -> lib_config:Lib_config.t
     -> instrument_with:Lib_name.t list
     -> unit
     -> t
 
   val find : t -> Lib_name.t -> lib option Memo.t
   val find_even_when_hidden : t -> Lib_name.t -> lib option Memo.t
-  val available : t -> Lib_name.t -> bool Memo.t
+  val find_sentinel : t -> Lib_info.Sentinel.t -> lib option Memo.t
+  val find_sentinel_even_when_hidden : t -> Lib_info.Sentinel.t -> lib option Memo.t
+  val available : t -> Lib_info.Sentinel.t -> bool Memo.t
+  val available_by_name : t -> Lib_name.t -> bool Memo.t
```

Comment:
You can flip these definitions around and preserve more existing code. E.g. `available` and `available_by_sentinel`.